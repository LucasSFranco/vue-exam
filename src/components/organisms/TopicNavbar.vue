<template>
  <AppContainer class="topic navbar">
    <h1 data-test="topic-navbar-title">
      {{ selectedTopic || 'all news' }}
    </h1>
    <nav>
      <router-link
        :class="{ 'selected': !selectedTopic }"
        :to="{ name: 'all-articles' }"
        data-test="topic-navbar-item"
      >
        all news
      </router-link>
      <router-link
        v-for="topic in topics"
        :key="topic"
        :class="{ 'selected': topic === selectedTopic }"
        :to="{
          name: 'topic-articles',
          params: { topic },
        }"
        data-test="topic-navbar-item"
      >
        {{ topic }}
      </router-link>
    </nav>
  </AppContainer>
</template>

<script>

import { mapState } from 'vuex'

import AppContainer from '@/components/atoms/AppContainer.vue'

export default {
  name: 'TopicNavbar',
  components: { AppContainer },
  computed: {
    ...mapState([
      'selectedTopic',
      'topics'
    ])
  }
}

</script>

<style lang="scss" scoped>

  .topic.navbar {
    margin-top: 5rem;

    h1 {
      color: var(--header-normal);
      font-family: 'Roboto', sans-serif;
      font-weight: 700;
      font-size: 2.625rem;
      margin-bottom: .75rem;
      text-transform: capitalize;
    }

    nav {
      border-top: 1px solid var(--accent-secondary);
      padding: .5rem 0;

      a {
        color: var(--text-primary);
        cursor: pointer;
        font-family: 'Roboto', sans-serif;
        font-size: .675rem;
        font-weight: 500;
        margin-bottom: .25rem;
        text-transform: uppercase;

        & + a {
          border-left: 1px solid var(--accent-secondary);
          margin-left: .5rem;
          padding-left: .5rem;
        }

        &:hover {
          text-decoration: underline;
        }

        &.selected {
          font-weight: 600;
        }
      }
    }
  }

</style>
