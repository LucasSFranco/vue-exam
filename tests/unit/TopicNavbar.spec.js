import { mount, RouterLinkStub } from '@vue/test-utils'

import TopicNavbar from '@/components/organisms/TopicNavbar.vue'

describe('Topic Navbar', () => {
  it('matches snapshot on root path', () => {
    const $store = {
      state: {
        selectedTopic: null,
        topics: ['science', 'technology']
      }
    }

    const wrapper = mount(TopicNavbar, {
      global: {
        mocks: { $store },
        stubs: {
          AppContainer: { template: '<slot />' },
          RouterLink: RouterLinkStub
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches snapshot on topic path', () => {
    const $store = {
      state: {
        selectedTopic: 'science',
        topics: ['science', 'technology']
      }
    }

    const wrapper = mount(TopicNavbar, {
      global: {
        mocks: { $store },
        stubs: {
          AppContainer: { template: '<slot />' },
          RouterLink: RouterLinkStub
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('properly changes url on nav-item click', () => {
    const $store = {
      state: {
        selectedTopic: null,
        topics: ['science', 'technology']
      }
    }

    const wrapper = mount(TopicNavbar, {
      global: {
        mocks: { $store },
        stubs: {
          AppContainer: { template: '<slot />' },
          RouterLink: RouterLinkStub
        }
      }
    })

    expect(wrapper.findAllComponents(RouterLinkStub)[0].props().to.name).toBe('all-articles')
    expect(wrapper.findAllComponents(RouterLinkStub)[1].props().to.name).toBe('topic-articles')
    expect(wrapper.findAllComponents(RouterLinkStub)[1].props().to.params.topic).toBe('science')
  })
})
