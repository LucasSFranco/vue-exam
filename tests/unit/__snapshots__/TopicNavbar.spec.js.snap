// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Navbar matches snapshot on root path 1`] = `
<h1 data-test="topic-navbar-title">all news</h1>
<nav><a class="selected" data-test="topic-navbar-item"> all news </a><a class="" data-test="topic-navbar-item">science</a><a class="" data-test="topic-navbar-item">technology</a></nav>
`;

exports[`Topic Navbar matches snapshot on topic path 1`] = `
<h1 data-test="topic-navbar-title">science</h1>
<nav><a class="" data-test="topic-navbar-item"> all news </a><a class="selected" data-test="topic-navbar-item">science</a><a class="" data-test="topic-navbar-item">technology</a></nav>
`;
